### PR TITLE
feat: atomsWithQueryAsync, plus example

### PR DESCRIPTION
**Pull request by Leweyse**
Fix to #21

### CODE REVIEW

**Comment by dai-shi** [comment]
Thanks for working on it. Can you please try adding some tests? You don't need to follow the patterns of existing tests.

File: src/atomsWithQueryAsync.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import type { QueryKey, QueryObserverOptions } from '@tanstack/query-core'
+import type { ExtractAtomValue, Getter, WritableAtom } from 'jotai'
```
Comment (dai-shi) [inline src/atomsWithQueryAsync.ts:2]: With https://github.com/pmndrs/jotai/pull/1741, we should be able to use it.
```suggestion
import type { ExtractAtomArgs, ExtractAtomValue, Getter, WritableAtom } from 'jotai'
```

**Comment by Leweyse** [comment]
Hi @dai-shi , I have a couple of doubts. 

First, I tried to add a test for the `refetch` functionality for the new async atom (`async query refetch` test), but I have some problems trying to set this up. Do you have any comments on that test? Any ideas on how can we approach it?

Second, when we fire the `dispatch` setter from the `atomsWithQueryAsync`, and we pass the force argument as `true`, the component is suspended for ever. You can find the code here: https://codesandbox.io/s/hopeful-silence-f2zs2k?file=/src/App.tsx

**Comment by dai-shi** [comment]
> First, I tried to add a test for the `refetch` functionality for the new async atom (`async query refetch` test), but I have some problems trying to set this up. Do you have any comments on that test? Any ideas on how can we approach it?

Not sure. Is it only about tests? It works in codesandbox, right?

> Second, when we fire the `dispatch` setter from the `atomsWithQueryAsync`, and we pass the force argument as `true`, the component is suspended forever. You can find the code here: https://codesandbox.io/s/hopeful-silence-f2zs2k?file=/src/App.tsx

Oh, I didn't realized that. Does it work with atomsWithQuery? Not sure if we can fix it. Maybe for now we reject using `force` in atomsWithQueryAsync?

**Comment by Leweyse** [comment]
I was able to fix the tests, and `force` has no effect on the query at this moment.

> > First, I tried to add a test for the `refetch` functionality for the new async atom (`async query refetch` test), but I have some problems trying to set this up. Do you have any comments on that test? Any ideas on how can we approach it?
> 
> Not sure. Is it only about tests? It works in codesandbox, right?
> 
> > Second, when we fire the `dispatch` setter from the `atomsWithQueryAsync`, and we pass the force argument as `true`, the component is suspended forever. You can find the code here: https://codesandbox.io/s/hopeful-silence-f2zs2k?file=/src/App.tsx
> 
> Oh, I didn't realized that. Does it work with atomsWithQuery? Not sure if we can fix it. Maybe for now we reject using `force` in atomsWithQueryAsync?

**Comment by dai-shi** [comment]
> I was able to fix the tests

Nice.

> and force has no effect on the query at this moment.

Fine. Do you want to disable the force option in atomsWithQueryAsync?

Please check CI results. Seems like trivial linting issue.

**Comment by Leweyse** [comment]
I'd rather not to disable it, so I worked on a fix. I updated the sandbox: https://codesandbox.io/s/hopeful-silence-f2zs2k?file=/src/App.tsx

**Pull request merged**